app/malleate_txs.go

```diff
@@ -47,7 +47,7 @@ func malleateTxs(
 	})
 
 	// split the tracked messagse apart now that we know the order of the indexes
-	msgs := make([]*core.Message, len(trackedMsgs))
+	msgs := make([]*core.Blob, len(trackedMsgs))
```

Comment:
[optional suggestion]
do you think using non pointers here (and ~20 lines above) would simplify things at all? for example, perhaps we could remove the extra conversion later in `PrepareProposal`

https://github.com/rootulp/celestia-app/blob/f888edab22c5553bb7b2a7d7c1806a567590016e/app/prepare_proposal.go#L44-L50

Reply:
Agreed. I took a stab at this before but I wasn't sure why `malleateTxs` returned a slice of pointers to messages instead of a slice of messages. Without unit tests on `malleateTxs` I'm hesitant to make changes but will try again.